TabnineSelfHosted/src/main/java/com/tabnineSelfHosted/chat/ChatEnabledState.kt

```diff
@@ -0,0 +1,57 @@
+package com.tabnineSelfHosted.chat
+
+import com.intellij.openapi.Disposable
+import com.intellij.util.messages.Topic
+import com.tabnineCommon.chat.ChatDisabledReason
+import com.tabnineCommon.chat.ChatFrame
+import com.tabnineCommon.chat.ChatState
+import com.tabnineCommon.general.StaticConfig
+import com.tabnineCommon.general.TopicBasedNonNullState
+import com.tabnineSelfHosted.binary.lifecycle.UserInfoStateSingleton
+import com.tabnineSelfHosted.binary.requests.userInfo.UserInfoResponse
+import java.util.function.Consumer
+
+class ChatEnabledState private constructor() : ChatFrame.UseChatEnabledState,
+    TopicBasedNonNullState<ChatState, ChatEnabledChanged>(
+        ENABLED_TOPIC, ChatState.loading()
+    ) {
+
+    companion object {
```

Comment:
why companion object? will the result be different from a static singleton?

Reply:
Companion object is basically a static instance. For the implementation of the class I always prefer not implementing statically. In general I don't really like singletons but the initialisation process is a little bit complex so it will do for now.